### ISSUE TITLE
SIGABRT crash after saving snapshot

### ISSUE THREAD

**Issue by zxcygnus** [mborik/GPMD85Emulator#12]
Emulator crashes immediatelly after any snapshot is saved. Snapshot was always created correctly and can be loaded without any problem.

I tried it on older Gentoo Linux and another computer with updated Debian 9 Jessie.

It doesn't depend on PMD configuration (selected model, or enabled peripherals) or running software, even after reset just in monitor it crashes same way.

[crash.txt](https://github.com/mborik/GPMD85Emulator/files/3007605/crash.txt)

**Comment by mborik** [comment]
Thank you Cygnus, great spot. I guess I've already met this issue years ago, but in my case it was occasional effect. Did you try it in debug or release (optimized) build?

**Comment by zxcygnus** [comment]
I am not sure if i understand correctly. I experimented with it a little, but i really don't know how to debug it (i know about gdb, but i am not experienced with it). I am trying everything on latest available version of emulator which is right now the version from 2018-03-26 - downloaded by git clone ...

I compiled emulator same way as you mentioned in INSTALL.md file. There was no issue, no errors or warnings during compilation. In Debian 9 is little older gcc 9.3.0-18, but it should work, right? SDL2 is in version 2.0.5. It is 64b system, but i have installed i386 libraries too. Same on Gentoo

On my retro PC with Debian 9 (9.8 more preciselly) i am running it under root right now, but on the same machine and somewhere else too i tried save snapshots under unpriviledged users too. There was no difference. I am just saying it, because there can be issue with permissions when saving file, but i doubt it is.

I am still not sure what you mean by debug or optimized build. I didn't use any parameters for ./configure i just compiled it and running it directly from that folder where binary was created.

Just for record, i can save memory dump or create PTP file without any problem. Only saving snapshots doesn't work.

And of course, i tried more hardware configurations of PMD, but i don't see difference, even in totally default configuration (PMD 85-2A without PMD 32-SD and memory expansion etc...) it behaves same.

Emulator doesn't crashes randomly, it crashes every time after snapshot is saved.
